### PR TITLE
adding File input component

### PR DESCRIPTION
**Pull request by A7madXatab**
Motive: the default file input component provided by browsers are not pleasing to the eye, most of the time it needs to be customized to make it viable within a design, what if we had a component that already took care of the designs for use?

this File input component solves this issue of the default styling, since it is now styles better than the default file input , and its a custom component itself, making the first of it's kind in the package

### CODE REVIEW

**Comment by A7madXatab** [comment]
i also have made changed into the website to incorporate it into the master branch when this component gets accepted :)

**Comment by driesvints** [comment]
Hey @A7madXatab. Can you please not open and close dozens of prs for the same thing? That just spams my notifications. Just push changes onto the same PR if you have any. 

I'll take a look at this soon.

**Comment by A7madXatab** [comment]
Hey @driesvints i do apologize for that, sorry for bothering with the PRs . thank you, wont happen again

File: resources/views/components/forms/inputs/file.blade.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<div>
+    <label for="file" style=" padding: .5rem; border-radius: 5px;" class="{{$buttonContainerClass}}">
```
Comment (bjuppa) [inline resources/views/components/forms/inputs/file.blade.php:2]: This `style` attribute makes the look of the component very opinionated, and css properties set in the `style` attribute can't be overridden by other CSS. So this locks the component into a single visual style. It's probably better than default browser file-inputs, but I think the purpose of this package is to provide a set of *functional* components, so ideally they wouldn't have any styling at all, or at least very easily configurable styling!

**Review by bjuppa — CHANGES_REQUESTED** [review]

**Pull request closed without merging**